hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java

```diff
@@ -319,10 +329,18 @@ public void closeCurrentAndSign() {
 
                 // create record file
                 try (FileOutputStream stream = new FileOutputStream(recordFile, false);
+                        GZIPOutputStream gzipStream =
+                                dynamicProperties.shouldCompressRecordFilesOnCreation()
+                                        ? new GZIPOutputStream(stream)
+                                        : null;
```

Comment:
A lot of duplicated conditional logic and unnecessary null checks. Should consolidate to a helper method. No need to flush file and gzip output streams separately since they will flush child streams.

```java
    private OutputStream createOutputStream(File file) {
        var fileOutputStream = new FileOutputStream(file, false);

        if (dynamicProperties.shouldCompressRecordFilesOnCreation()) {
            return new GZIPOutputStream(fileOutputStream);
        }

        return fileOutputStream;
    }
```

```suggestion
                try (var stream = createOutputStream(recordFile);
```

Reply:
Yeah, was wondering if we can make this cleaner as well...
Took the current approach, because we have  
```
stream.getChannel().force(true);
stream.getFD().sync();
```
after flushing, and these methods are defined in `FileOutputStream` (which isn't part of `GZIPOutputStream's` class hierarchy), so we cannot generalize to `OutputStream`.

Reply:
You could extend `FileOutputStream` as a `FileSyncOutputStream` and do the flushing in the close method so it's done automatically. Then can do above with gzip.

Reply:
BTW, it's unrelated to your PR, but it looks like it's [unnecessary](https://stackoverflow.com/questions/5650327/are-filechannel-force-and-filedescriptor-sync-both-needed) to call both `stream.getChannel().force(true)` and `stream.getFD().sync()` as they both seem to call `fsync` internally. Calling `fsync` twice per file increases disk I/O unnecessarily.

Reply:
Hmm, maybe we can log this as an issue and look into it after we have all the RSV6/Traceability testing in place, on the off chance this breaks something?

Reply:
Opened #3849